app/script/auth/AuthError.js

```diff
@@ -23,37 +23,33 @@ window.z = window.z || {};
 window.z.auth = z.auth || {};
 
 z.auth.AuthError = class AuthError extends Error {
-  constructor(type) {
+  constructor(type, message) {
     super();
 
     this.name = this.constructor.name;
     this.stack = new Error().stack;
     this.type = type || AuthError.TYPE.UNKNOWN;
 
-    switch (this.type) {
-      case AuthError.TYPE.COOKIES_DISABLED:
-        this.message = 'Cookies are disabled';
-        break;
-      case AuthError.TYPE.INDEXED_DB_UNSUPPORTED:
-        this.message = 'IndexedDB is not supported';
-        break;
-      case AuthError.TYPE.MULTIPLE_TABS:
-        this.message = 'Cannot open in multiple tabs simultaneously';
-        break;
-      case AuthError.TYPE.PRIVATE_MODE:
-        this.message = 'Unsupported Private Mode';
-        break;
-      default:
-        this.message = 'Unknown AuthError';
-    }
+    this.message = message || AuthError.MESSAGE[this.type] || AuthError.MESSAGE.UNKNOWN;
```

Comment:
I see this pattern quite some times in this PR. Maybe having a super class that does that could be a good idea. 
How about a `WireError` and make all of those inherit from it?
WDYT?

Reply:
Yeah. Sounds like a good idea.